src/pykeen/evaluation/rank_based_evaluator.py

```diff
@@ -235,160 +482,18 @@ def resolve_metric_name(name: str) -> MetricKey:
     rank_type = RANK_TYPE_SYNONYMS.get(rank_type, rank_type)
     if rank_type not in RANK_TYPES:
         raise ValueError(f"Invalid rank type: {rank_type}. Allowed are {RANK_TYPES}.")
-    elif rank_type != RANK_REALISTIC and name in TYPES_REALISTIC_ONLY:
+    if rank_type != RANK_REALISTIC and name in TYPES_REALISTIC_ONLY:
         raise ValueError(f"Invalid rank type for {name}: {rank_type}. Allowed type: {RANK_REALISTIC}")
 
-    return MetricKey(name, side, rank_type, k)
+    return MetricKey(name, side, rank_type, k)  # type: ignore
 
 
-@fix_dataclass_init_docs
-@dataclass_json
-@dataclass
 class RankBasedMetricResults(MetricResults):
     """Results from computing metrics."""
 
-    arithmetic_mean_rank: Dict[str, Dict[str, float]] = field(
-        metadata=dict(
-            name="Mean Rank (MR)",
-            increasing=False,
-            range="[1, inf)",
-            doc="The arithmetic mean over all ranks.",
-            link="https://pykeen.readthedocs.io/en/stable/tutorial/understanding_evaluation.html#mean-rank",
```

Comment:
some of this metadata is currently lost, and should go as docdata into the docstrings - @cthoyt how should this be added?